### PR TITLE
Document validators

### PR DESCRIPTION
**Pull request by amayer5125**
Simplify the code a little to make it easier to read.

### CODE REVIEW

File: src/Listener/JsonApi/DocumentValidator.php [hunk 1]
```diff
@@ -290,14 +290,11 @@ protected function _primaryDataMayHaveRelationships(): bool
             // multiple hasMany relationships
             $hasManys = $this->_getProperty($relationshipPathObject->dotted . '.data');
 
-            $i = 0;
```
Comment (geoidesic) [inline src/Listener/JsonApi/DocumentValidator.php:293]: Thanks for your PR!

The previous file's change I agree with. This one not. This isn't an improvement. `foreach` is slower than `for` loop and it's really not any more difficult to read. This is purely a stylistic preference on your part.

**Review by geoidesic — CHANGES_REQUESTED** [review]
Please revert changes in `DocumentValidator.php`. The ones in DocumentRelationshipValidator.php` is good. Tx

File: src/Listener/JsonApi/DocumentValidator.php (hunk 1 above)
Comment (amayer5125) [reply src/Listener/JsonApi/DocumentValidator.php:293]: If `foreach` is slower, why would we stick with the slower implementation? My implementation uses `for` which you say is faster.

If you still want me to revert the changes in this file I will do that, but I feel what I have here is better.

Comment (geoidesic) [reply src/Listener/JsonApi/DocumentValidator.php:293]: No, you're 100% right, I was looking at the PR in reverse. Too little sleep last night LOL

Comment (amayer5125) [reply src/Listener/JsonApi/DocumentValidator.php:293]: Haha. No worries, I get it.

**Pull request merged**